posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,452 @@
++
+[source, java]
+----
+@ConfigProperties(prefix="customer")
+@Dependent
+public static class BeanOne {
+   private String name;
+   int age;
+}
+----
++
+The code above retrieves the following defined properties for the `name` and `age` fields:
++
+[source, java]
+----
+customer.name=Bob
+customer.age=24
+----
+
+* Allows you to easily determine which was the "winning" `ConfigSource` (the `ConfigSource` with the highest ordinal) for a specified property via link:https://github.com/eclipse/microprofile-config/blob/master/api/src/main/java/org/eclipse/microprofile/config/ConfigValue.java[ConfigValue]. A `ConfigValue` object holds information about the configuration property name, configuration value, the ConfigSource name from where the configuration property was loaded and the ordinal of the `ConfigSource`. You can access these values with the following method:
++
+[source, java]
+----
+ConfigValue cv = Config.getConfigValue(String propertyName);
+cv.getSourceName() // the config source that supplies the configure value for the above property
+----
+
+* Enables an application to include configuration for different environments and development stages while only one of them is active (e.g. `dev`, `test`, `prod`) via Configuration Profiles. The Config profile can be specified via the `mp.config.profile` property, which can be set in any of the configuration sources. Once it is set, the corresponding properties associated with the active profile are used. The value can also be set at application start-up, for example:
++
+[source, xml]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
++
+The `mp.config.profile` property works on both the property level and config file level:
++
+On the property level profiles can be utilised by setting Config property values specific to the selected profile. For example, the `vehicle.name` property in the following example would be set to `bike` from the previous Config start up command:
++
+[source, java]
+----
+%dev.vehicle.name=car
+%testing.vehicle.name=bike
+vehicle.name=lorry
+----
++
+If `mp.config.profile` was not set, `vehicle.name` would be set to `lorry`.
++
+On the config file (`microprofile-config.property`) level the property `mp.config.profile` also activates the corresponding `microprofile-config-<mp.config.profile>.properties` with `microprofile-config.properties` as the fallback.
+
+* Supports link:https://download.eclipse.org/microprofile/microprofile-config-2.0-RC1/microprofile-config-spec.html#property-expressions[Property Expressions], which provide a way to set and expand variables in property values. For example, in the following configuration properties, the property `server.url` will be resolved to `http://example.org/endpoint`:
++
+[source, java]
+----
+server.url=http://${server.host}/endpoint
+server.host=example.org
+----
++
+*Note*: Previous working configurations may behave differently (if configuration in use contains values with Property Expressions syntax).
```

Comment:
Change to "Previous working configurations might behave differently if the configuration in use contains values with Property Expressions syntax."